src/Processors/Executors/PullingPipelineExecutor.cpp

```diff
@@ -44,6 +44,9 @@ bool PullingPipelineExecutor::pull(Chunk & chunk)
     if (!executor)
         executor = std::make_shared<PipelineExecutor>(pipeline.processors, pipeline.process_list_element);
 
+    if (!executor->checkTimeLimit())
```

Comment:
Note: if we throw exception here, executor is not stopped (until the moment of ~PullingPipelineExecutor)

Reply:
I understand that's not an issue, right? If nobody catches the exception the executor destructor will be called, if somebody does then it will need to handle it and eventually the executor will be cancelled.

Reply:
It is still not very cool. E.g. executor itself may continue working (e.g. in case of bug)., and we may wait for a long time in destructor.